File: Leetcode/Daily/06-2025/TypeScript/18.ts
// Problem: 2966. Divide Array Into Arrays With Max Difference (Medium)
// Link: https://leetcode.com/problems/divide-array-into-arrays-with-max-difference

function divideArray(nums: number[], k: number): number[][] {
    let n = nums.length;
    let ans: number[][] = [];

    nums.sort((a, b) => a-b);
    for (let i = 0; i < n; i += 3) {
        if (nums[i+2] - nums[i] > k) return [];
        ans.push([nums[i], nums[i+1], nums[i+2]]);
    }

    return ans;
};

File: Leetcode/Daily/06-2025/Python3/18.py
# Problem: 2966. Divide Array Into Arrays With Max Difference (Medium)
# Link: https://leetcode.com/problems/divide-array-into-arrays-with-max-difference

class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        
        n = len(nums)
        ans = []
        nums.sort()

        i = 0

        while i < n:
            if nums[i+2] - nums[i] > k:
                return []
            ans.append([nums[i], nums[i+1], nums[i+2]])

            i += 3

        return ans;        